kolibri/plugins/learn/assets/src/views/CustomChannelRenderer.vue

```diff
@@ -0,0 +1,202 @@
+<template>
+
+  <div>
+    <KContentRenderer
+      ref="customiframe"
+      class="content-renderer"
+      :kind="topic.kind"
+      :lang="topic.lang"
+      :files="topic.files"
+      :options="topic.options"
+    />
+    <!-- placeholder for jb's work on custom overlay -->
+    <KModal
+      v-if="open"
+      :title="currentContent.title"
+      cancelText="Close"
+      @cancel="open = false"
+    >
+      <KContentRenderer
+        ref="customiframe"
+        class="content-renderer"
+        :kind="currentContent.kind"
+        :lang="currentContent.lang"
+        :files="currentContent.files"
+        :options="currentContent.options"
+      />
+    </KModal>
+  </div>
+
+</template>
+
+
+<script>
+
+  import pick from 'lodash/pick';
+  import { mapState } from 'vuex';
+  import { ContentNodeResource } from 'kolibri.resources';
+  import Hashi from 'hashi';
+  import router from 'kolibri.coreVue.router';
+
+  export default {
+    name: 'CustomChannelRenderer',
+    data() {
+      return {
+        currentContent: this.topic,
+        open: false,
+      };
+    },
+    computed: {
+      ...mapState('topicsTree', ['topic']),
+      context() {
+        const urlParams = new URLSearchParams(window.location.search);
+        const fetchedEncodedContext = urlParams.has('context')
```

Comment:
Context is showing up like this in the URLS: `context=%5Bobject%20Object%5D`. You might need to `JSON.stringify` before passing it out. Similarly will need to parse after decoding it.

Reply:
You can also retrieve the params from `this.$route.params` rather than the global location search.